src/oscar/views/generic.py

```diff
@@ -172,7 +172,7 @@ def __init__(self, *args, **kwargs):
             number_required = False
             number_help_text = ''
 
-        self.fields['phone_number'] = forms.CharField(
+        self.fields['phone_number'] = forms.CharField(label=_('Phone number'),
```

Comment:
I think we need to inherit the label from the model field if it already exists - i.e., we need to set the label in the try/catch block above, and only set it again if we didn't find one.